src/api/app/components/sign_up_component.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+class SignUpComponent < ApplicationComponent
+  attr_accessor :submit_btn_text
+
+  def initialize(submit_btn_text:)
```

Comment:
_Nitpick_

Instead of doing `@submit_btn_text = submit_btn_text || 'Sign up'` on line 8, I suggest this:
```suggestion
  def initialize(submit_btn_text: 'Sign up')
```

Then the line 8 would be `@submit_btn_text = submit_btn_text`.

Reply:
That's why my first try, but sometimes we pass to the component params[:submit_btn_text] which can be nil, look https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/webui/users_controller.rb#L45

Reply:
> I would prefer we get rid of passing this via `params`

I agree with you. If you can get rid of this, it's great!